README.md

```diff
@@ -158,3 +160,67 @@ const validators = [
 // attaching our Form to form-container with validation
 export default connectForm(validators)(Form);
 ```
+
+### Submit validation example
+
+Form Container exposes `SubmissionError`, a throwable error which can be used to set submit validation for fields in response to promise rejection because of validation errors.
+
+A wrapped form component has access to `handleSubmit` via `formMethods`:
+
+### `handleSubmit([submit])`
+
+#### Arguments
+
+*   [`submit: (model) => Promise`]: A submit handler function which has access to the form model and returns a Promise.
+
+Inside the provided `submit` function you can throw a `SubmissionError` which will be caught by `handleSubmit`, and the submission errors will be set for each key in the form state under a key `submitErrors`, e.g.:
+
+```javascript
+throw new SubmissionError({
+    test: "Don't like this..."
+});
+```
+
+```javascript
+// components/Form.tsx
+import * as React from 'react';
+import { connectForm, ValidationRuleFactory, Condition, IFormProps } from 'form-container';
+
+interface IProps extends IFormProps {}
+
+// arbitrary form component
+export class Form extends React.PureComponent<IProps, {}> {
+    submit = (model: any) => {
```

Comment:
do we know the shape of the model?

Reply:
Good shout. I was just going along with the definition of model as `any` in `IFormProps`, but can define an interface and pass as generic to `connectForm`

Reply:
ok, let's leave it as is for now